x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/curation_logic.ts

```diff
@@ -235,6 +236,8 @@ export const CurationLogic = kea<MakeLogicType<CurationValues, CurationActions,
           `/internal/app_search/engines/${engineName}/curations/${props.curationId}`,
           { query: { skip_record_analytics: 'true' } }
         );
+        response.hidden = response.hidden.map((x) => flattenObject(x) as CurationResult);
```

Comment:
🤦 I don't have enough knowledge of TypeScript to rewrite this with generics. I need some handholding from frontend here.

Reply:
So I tried to actually do this just now, and it doesn't work because TypeScript is annoying about dictionary-defined objects. So what you're doing here is...fine! Nevermind my comment about generics.

I do have a new question now: you're using CurationResult here, which has `id: string`, but Result below, which has `id: { raw: string}`. Is that correct?

Reply:
Hmm... seems to work... not sure why. 🤷